### PR TITLE
Upload files to storage bucket directly, not through references

### PR DESCRIPTION
**Pull request by ccanos**
### Describe the background of your pull request

What does your pull request do? Does it solve a bug (which one?), add a feature?

### Additional context

Add any other context

### Governance 

- [ ] Documentation is added
- [ ] Test cases are added or updated

By submitting this pull request I confirm that:
- I've read the contributing document https://github.com/alkem-io/.github/blob/master/CONTRIBUTING.md.
- I've read and understand the Code of Conduct https://github.com/alkem-io/.github/blob/master/CODE_OF_CONDUCT.md.
- I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/alkem-io/Coordination/blob/master/LICENSE.
- I understand that submitting the pull request requires agreeing to and signing the contributor license agreement.

### CODE REVIEW

File: src/domain/storage/storage-bucket/storage.bucket.resolver.mutations.ts [hunk 1]
```diff
@@ -170,4 +171,49 @@ export class StorageBucketResolverMutations {
     };
     return await this.referenceService.updateReference(updateData);
   }
+
+  @UseGuards(GraphqlGuard)
+  @Mutation(() => String, {
+    description:
+      'Create a new Document on the Storage and return the value as part of the returned Reference.',
```
Comment (hero101) [inline src/domain/storage/storage-bucket/storage.bucket.resolver.mutations.ts]: fix the description

File: src/domain/storage/storage-bucket/storage.bucket.resolver.mutations.ts [hunk 2]
```diff
@@ -170,4 +171,49 @@ export class StorageBucketResolverMutations {
...
+
+    const documentAuthorized =
+      await this.documentAuthorizationService.applyAuthorizationPolicy(
+        document,
+        storageBucket.authorization
+      );
+
+    const uri =
+      this.documentService.getPubliclyAccessibleURL(documentAuthorized);
+    return uri;
```
Comment (hero101) [inline src/domain/storage/storage-bucket/storage.bucket.resolver.mutations.ts]: return it inline

File: src/domain/storage/storage-bucket/storage.bucket.resolver.mutations.ts [hunk 3]
```diff
@@ -170,4 +171,49 @@ export class StorageBucketResolverMutations {
...
+  ): Promise<string> {
+    const storageBucket =
+      await this.storageBucketService.getStorageBucketOrFail(
+        uploadData.storageBucketId
+      );
+
+    this.authorizationService.grantAccessOrFail(
+      agentInfo,
+      storageBucket.authorization,
+      AuthorizationPrivilege.READ, //FILE_UPLOAD,
```
Comment (hero101) [inline src/domain/storage/storage-bucket/storage.bucket.resolver.mutations.ts]: This has to be addressed on policy level I think with the proper privilege.

**Review by hero101 — CHANGES_REQUESTED** [review]
Great overall. Tested locally.
Minor adjustments are needed.

File: src/domain/storage/storage-bucket/storage.bucket.resolver.mutations.ts (hunk 3 above)
Comment (hero101) [reply src/domain/storage/storage-bucket/storage.bucket.resolver.mutations.ts]: @techsmyth We need to discuss how is this going to be achieved. We need an extra credential rule to apply FILE_UPLOAD on the bucket if the user has a membership to the parent journey.

Comment (techsmyth) [reply src/domain/storage/storage-bucket/storage.bucket.resolver.mutations.ts]: Agreed, but needs a bit of discussion about how to assign that privilege as do now want that the privilege shows up everywhere in the tree as it is very specific to the storage bucket entity. Suggest we chat it through...

One option would be to assign it in the following location(so line 422):
image

+ then raise another issue for fixing that the FILE_UPLOAD is applied too widely. So as to not hold this issue up.

**Review by hero101 — APPROVED** [review]
Tested locally, with the latest rule changes.
Auth reset is required.

File: src/domain/storage/storage-bucket/storage.bucket.resolver.mutations.ts (hunk 3 above)
Comment (hero101) [reply src/domain/storage/storage-bucket/storage.bucket.resolver.mutations.ts]: Resolved with https://github.com/alkem-io/server/pull/2956

**Pull request merged**